src/audio/google_rtc_audio_processing.c

```diff
@@ -34,8 +34,18 @@
 
 #include <google_rtc_audio_processing.h>
 #include <google_rtc_audio_processing_platform.h>
+#include <google_rtc_audio_processing_sof_message_reader.h>
 
-#define GOOGLE_RTC_AUDIO_PROCESSING_SAMPLERATE 48000
+#define SAMPLE_RATE_HZ 48000
+#define SAMPLES_PER_CHANNEL (SAMPLE_RATE_HZ / 100)
+#define DEFAULT_NUM_AEC_REFERENCE_CHANNELS 2
+#define DEFAULT_NUM_CAPTURE_INPUT_CHANNELS 1
+#define DEFAULT_NUM_CAPTURE_OUTPUT_CHANNELS 1
+
+#define MEMORY_BUFFER_STEREO_SIZE_BYTES 250000
+#define MEMORY_BUFFER_MONO_SIZE_BYTES 200000
+#define MEMORY_BUFFER_BYTES (DEFAULT_NUM_CAPTURE_OUTPUT_CHANNELS == 1 ? \
```

Comment:
Yes. But my concern with that is that this size needs to be defined compile-time, so if a macro is defined, it mustn't be used on any runtime-specified parameters. 
With that I'm not convinced that adding a macro is the correct thing to do.

Reply:
You could add a complie time assert on the final calculated value to make sure it is compile time available. Also why does it need to be compile time ready?

Reply:
Yes, I think you are right, and my reasons for keeping this as a #define does not really hold.
I changed it to a function instead though (because I would prefer to not use a macro).

The information needs to be available at the call to _create though. However, if the buffer allocated is too small, the functionality fallbacks to using the SOF allocation routines.

Reply:
My biggest issue is that it detaches the pipeline expectations from the component. I think it would be wiser to have the component check the pipeline against its needs rather than assume based on a compilation config

Reply:
That being said, we do have the issue right now where DMIC is more channels than we actually care about. Might be worth fixing up a demux properly. @sathya-nujella FYI

Reply:
The scheme here is fairly easily extended to react on the pipeline needs.
Right now, however, the component _copy code is so hardcoded to always process mono mic channels, so I'd prefer to postpone that change until the _copy code is properly extended.